### PR TITLE
#19157 now metadata is available only for admin

### PR DESCRIPTION
**Pull request by jdotcms**
The metadata on saml is only available for admin users

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 1]
```diff
@@ -232,6 +236,13 @@ public void metadata( @PathParam( "idpConfigId" ) final String idpConfigId,
 						  @Context final HttpServletRequest httpServletRequest,
 						  @Context final HttpServletResponse httpServletResponse ) throws IOException {
 
+		if (!new WebResource.InitBuilder(this.webResource).rejectWhenNoUser(true).
+				requestAndResponse(httpServletRequest, httpServletResponse)
+				.requiredBackendUser(true).init().getUser().isAdmin()) {
+
+			throw new RuntimeException(new DotSecurityException("SAML metadata is only accessible by administrators"));
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:243]: Codacy found an issue: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

**Review by nollymar — COMMENTED** [review]
An IT would be great on this scenario

**Comment by dsilvam** [comment]
agree on the missing test

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 2]
```diff
@@ -232,6 +251,13 @@ public void metadata( @PathParam( "idpConfigId" ) final String idpConfigId,
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:258]: Codacy found an issue: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 3]
```diff
@@ -65,6 +69,21 @@ public DotSamlResource() {
 		this.samlAuthenticationService            = DotSamlProxyFactory.getInstance().samlAuthenticationService();
 		this.identityProviderConfigurationFactory = DotSamlProxyFactory.getInstance().identityProviderConfigurationFactory();
 		this.samlHelper                           = new SAMLHelper(this.samlAuthenticationService);
+		this.webResource						  = new WebResource();
+	}
+
+	@VisibleForTesting
+	protected DotSamlResource(final SamlConfigurationService           samlConfigurationService,
+							final SAMLHelper           				   samlHelper,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:77]: Codacy found an issue: [Avoid unused constructor parameters such as 'samlHelper'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 4]
```diff
@@ -65,6 +69,21 @@ public DotSamlResource() {
...
+							final SamlAuthenticationService            samlAuthenticationService,
+							final IdentityProviderConfigurationFactory identityProviderConfigurationFactory,
+							final WebResource						   webResource) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:80]: Codacy found an issue: [Avoid unused constructor parameters such as 'webResource'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 5]
```diff
@@ -65,6 +69,21 @@ public DotSamlResource() {
 		this.samlAuthenticationService            = DotSamlProxyFactory.getInstance().samlAuthenticationService();
 		this.identityProviderConfigurationFactory = DotSamlProxyFactory.getInstance().identityProviderConfigurationFactory();
 		this.samlHelper                           = new SAMLHelper(this.samlAuthenticationService);
+		this.webResource						  = new WebResource();
+	}
+
+	@VisibleForTesting
+	protected DotSamlResource(final SamlConfigurationService           samlConfigurationService,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:76]: Codacy found an issue: [Avoid unused constructor parameters such as 'samlConfigurationService'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 6]
```diff
@@ -65,6 +69,21 @@ public DotSamlResource() {
...
+							final SAMLHelper           				   samlHelper,
+							final SamlAuthenticationService            samlAuthenticationService,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:78]: Codacy found an issue: [Avoid unused constructor parameters such as 'samlAuthenticationService'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.mockito.Mockito.mock;
+
+public class DotSamlResourceTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java:25]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * Method to test: testing the metadata of the DotSamlResource
+     * Given Scenario: the test creates a limited user and call the saml metadata
+     * ExpectedResult: Must throw a security exception
+     *
+     */
+    @Test(expected = SecurityException.class)
+    public void testMetadata() throws DotDataException, DotSecurityException, IOException {
+
+        final User limitedUser = mock(User.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java:40]: Codacy found an issue: [Avoid unused local variables such as 'limitedUser'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 9]
```diff
@@ -65,6 +69,21 @@ public DotSamlResource() {
...
+							final IdentityProviderConfigurationFactory identityProviderConfigurationFactory,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:79]: Codacy found an issue: [Avoid unused constructor parameters such as 'identityProviderConfigurationFactory'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java:25]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java:40]: Codacy found an issue: [Avoid unused local variables such as 'limitedUser'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:258]: Codacy found an issue: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:258]: Codacy found an issue: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DotSamlResourceTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java:27]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:258]: Codacy found an issue: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

File: dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/auth/providers/saml/v1/DotSamlResourceTest.java:27]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077896)

**Pull request merged**